src/uniformscalingmap.jl

```diff
@@ -6,12 +6,16 @@ struct UniformScalingMap{T} <: LinearMap{T}
         return new{typeof(λ)}(λ, M)
     end
 end
-UniformScalingMap(λ::Number, M::Int, N::Int) =
-    (M == N ?
-        UniformScalingMap(λ, M) : error("UniformScalingMap needs to be square"))
-UniformScalingMap(λ::T, sz::Dims{2}) where {T} =
-    (sz[1] == sz[2] ?
-        UniformScalingMap(λ, sz[1]) : error("UniformScalingMap needs to be square"))
+@deprecate UniformScalingMap(λ::Number, M::Int, N::Int) UniformScalingMap(λ, M) false
+@deprecate UniformScalingMap(λ::Number, sz::Dims{2}) UniformScalingMap(λ, sz[1]) false
+# the following methods are misleading: they introduce the opportunity to pass 2 dims,
+# but throw if the dims are unequal!
+# UniformScalingMap(λ::Number, M::Int, N::Int) =
+#     (M == N ?
+#         UniformScalingMap(λ, M) : error("UniformScalingMap needs to be square"))
+# UniformScalingMap(λ::T, sz::Dims{2}) where {T} =
+#     (sz[1] == sz[2] ?
+#         UniformScalingMap(λ, sz[1]) : error("UniformScalingMap needs to be square"))
 
```

Comment:
```suggestion
Base.getindex(A::UniformScalingMap, args...) = A.λ * A.lmap.getind(args...)
```
Seems like something like this should be the natural default for this type.